### ISSUE TITLE
No error backpropagation through skip connections

### ISSUE THREAD

**Issue by FelixGruen** [jakeret/tf_unet#7]
Hi,

[`tf.image.extract_glimpse()`](https://github.com/jakeret/tf_unet/blob/master/tf_unet/layers.py#L51) is used in the implementation of the [`crop_and_concat()`](https://github.com/jakeret/tf_unet/blob/master/tf_unet/layers.py#L47) function.

`tf.image.extract_glimpse()` does not allow for backpropagation of gradients (if it was the only connection between two nodes the gradient returned by TensorFlow would be `None`) and the gradients with respect to the segmentation error for all layers before a skip connection will therefore be off.

[`tf.slice()`](https://www.tensorflow.org/versions/r0.12/api_docs/python/array_ops.html#slice) should be used instead.

**Comment by jakeret** [comment]
Good to know, i wasn't aware of that. If you have time I would really appreciate a PR ;-)

**Comment by FelixGruen** [comment]
I sent you a PR just now :)

You should make sure it doesn't lead to any conflicts since I changed `crop_and_concat(x1,x2,output_shape)` to `crop_and_concat(x1,x2)`. As far as I could see there was only one call to the function and I fixed that, but you probably know the code better than I do.

**Comment by jakeret** [comment]
Sweet! I'll have a closer look soon.
Have you checked if it works when the shape of the tensor is not even? I'm asking because I had an issue in a image cropping function

**Comment by Mottotime** [comment]
IMHO, there should not be gradient flow from the skip connection. This may cause optimisation problems.

**Comment by jakeret** [comment]
Could you elaborate a bit more?
